snapcraft/internal/meta/package_repository.py

```diff
@@ -181,40 +236,119 @@ def marshal(self) -> Dict[str, Any]:
             data["key-server"] = self.key_server
 
         data["name"] = self.name
-        data["suites"] = self.suites
+
+        if self.path:
+            data["path"] = self.path
+
+        if self.suites:
+            data["suites"] = self.suites
+
         data["url"] = self.url
 
         return data
 
-    def validate(self) -> None:
-        if self.formats is not None and any(
-            f not in ["deb", "deb-src"] for f in self.formats
-        ):
-            raise errors.PackageRepositoryValidationError(
-                url=self.url,
-                brief=f"Invalid formats {self.formats!r}.",
-                resolution="You can specify a list of formats including 'deb' and/or 'deb-src'.",
-            )
+    def validate(self) -> None:  # noqa: C901
+        if self.architectures is not None:
```

Comment:
valid architectures are dependant on base, core20 and i386 are invalid as is core18 or less with riscv

Reply:
This might be a fragile check, might be worth just letting apt fail and surface an error

Reply:
Good point.   It was mostly meant as a sanity check, riscv enablement would require it to be added here.

Longer term I'd hope for a shared deb arch enum that we could check against.

I can remove it if you want.

Reply:
All our enums are in ProjectOptions today, maybe add it there?

Reply:
Then again, it is not uncommon for an arch to be enabled post release, so this would require doing post facto work, removing might just be the better option if we can surface the error correctly

Reply:
Removed.